services/repository/branch.go

```diff
@@ -463,15 +466,21 @@ var (
 	ErrBranchIsDefault = errors.New("branch is default")
 )
 
-// DeleteBranch delete branch
-func DeleteBranch(ctx context.Context, doer *user_model.User, repo *repo_model.Repository, gitRepo *git.Repository, branchName string) error {
-	err := repo.MustNotBeArchived()
+func CanDeleteBranch(ctx context.Context, repo *repo_model.Repository, branchName string, doer *user_model.User) error {
```

Comment:
Maybe it would be better to return a boolean type

Reply:
It returns an error so that we know the reason why it can not delete branch.